src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -205,32 +220,35 @@ public function getMarkingStore()
         return $this->markingStore;
     }
 
+    /**
+     * @param object     $subject
+     * @param Marking    $marking
+     * @param Transition $transition
+     *
+     * @return TransitionBlockerList
+     */
     private function doCan($subject, Marking $marking, Transition $transition)
     {
         foreach ($transition->getFroms() as $place) {
             if (!$marking->has($place)) {
-                return false;
+                return new TransitionBlockerList(array(TransitionBlocker::createNotApplicable($transition->getName())));
```

Comment:
I have an issue this this new feature.
If the a transition is not enabled because of a missing place in a marking, the end user will only know the **first** reason. It's the same for the guard, they will not be triggered.

Anyway, it's a very bad idea to trigger the guard listener if the marking is not valid (performance issue).

What do you think about that?

Reply:
You're right in saying, that the end user won't know all the blocker reasons, if the the subject isn't in the required state to start with (i.e. the marking is missing a place). I'd say: this is not a bug, but a feature.

I certainly don't need to know the rest of the reasons, if the subject fails to be in the correct state to begin with. And, as you correctly mentioned, it'd be a (massive) performance issue to run the guards in this case, so that's another reason for not doing it.

So what I think is this:
1. Ain't nobody gonna need this (yagni).
2. It's a performance issue.
3. If someone needs this, then he needs to be explained, that this would be a performance issue and therefore it won't be implemented.